core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -160,6 +160,9 @@ class BaseUnit : INamed, IConstruction {
             && construction.cityInfo.containsBuildingUnique("All newly-trained melee, mounted, and armored units in this city receive the Drill I promotion"))
             unit.promotions.addPromotion("Drill I", isFree = true)
 
+        //movement penalty
+        if(!unit.hasUnique("Starts with no movement penalty"))
+        unit.currentMovement = 0f
```

Comment:
What if I build a unit, not buy it? Should it also be built with a movement penalty?

Reply:
thats a good point, for this I think there is no way than to refactor the postBuildEvent function and give it a parameter like "wasBought: Boolean" with a defaultvalue of false